node/src/types/block.rs

```diff
@@ -1751,15 +2046,9 @@ mod tests {
         let bogus_block_hash = hash::hash(&[0xde, 0xad, 0xbe, 0xef]);
         block.header.body_hash = bogus_block_hash;
 
-        let actual_body_hash = block.body.hash();
-
         // No Eq trait for BlockValidationError, so pattern match
-        match block.verify() {
-            Err(BlockValidationError::UnexpectedBodyHash {
-                expected_by_block_header,
-                actual,
-            }) if expected_by_block_header == bogus_block_hash && actual == actual_body_hash => {}
-            unexpected => panic!("Bad check response: {:?}", unexpected),
+        if block.verify().is_ok() {
+            panic!("Block has bogus body hash: {:?}", block)
```

Comment:
Why not compare the expected error, as in the old code?

Reply:
Good question. I have no idea why this change happened. It's present on the `feat-fast-sync` branch, though. @xcthulhu Any ideas?

Reply:
We could revert this to not drop that message, I'm fine with that.